### ISSUE TITLE
integrate DiffRules

### ISSUE THREAD

**Issue by CarloLucibello** [denizyuret/AutoGrad.jl#51]
it would be good to slim down autograd using DiffRules.jl:
https://github.com/JuliaDiff/DiffRules.jl/blob/master/src/rules.jl

**Comment by denizyuret** [comment]
Agreed.  Let's make this part of the julia-0.7 roadmap as well (a Cassette based solution might be in the works but unlikely to be ready before julia-1.1).
